Implement a --verbose flag

bernhard-herzog opened the pull request:
When fakedoc is invoked with the --verbose flag, some additional information is printed to stdout:

 - Which files are being generated

 - Which files are read: which template, files read for book elements

The interaction with other outputs, like the [n/m] progress indicator or printing the random number generator seed is not optimal, but it's also not so clear, just how they should interact.

bernhard-herzog committed:
Implement a --verbose flag

When fakedoc is invoked with the --verbose flag, some additional
information is printed to stdout:

 - Which files are being generated

 - Which files are read: which template, files read for book elements

The interaction with other outputs, like the [n/m] progress indicator or
printing the random number generator seed is not optimal, but it's also
not so clear, just how they should interact.

bernhardreiter commented on cmd/fakedoc/main.go:
@bernhard-herzog did you see the revive warning for line 37?

bernhardreiter commented:
@cintek can you try it and see if it runs to your expectation.

cintek reviewed (changes requested):
When I tested fakedoc with type `book` this was the result:

```
Loading template "template.toml"
pcg:20127943b45ce895:abb9d8c29c6e7f31
[5/100]generating "generated_docs/csaf-4.json"Loading book "lorem.txt"
[100/100]generating "generated_docs/csaf-99.json"
```

I would prefer to move the "Loading book" entry to a separate line so it is not so easy to miss it. By adding the progress status in front of it users can still see where the entry belongs to.

```
[5/100]generating "generated_docs/csaf-4.json"
[5/100]Loading book "lorem.txt"
[100/100]generating "generated_docs/csaf-99.json"
```

cintek commented:
> The interaction with other outputs, like the [n/m] progress indicator or printing the random number generator seed is not optimal

This seems okay for me because the users have at least one example where to look for the file and because of the progress indicator they should be able to understand that the rest will look similar :)

bernhard-herzog committed:
Tweak newline handling in --verbose output

"Loading book" now appears on separate lines. The progress indicators
and "generating .." lines do not have newlines so that the carriage
returns still work.

When generating a single file, the verbose output has a newline at the
end now. There are cases where this is one newline too many, but this
seems better than having one too few. This could be avoided, but would
require more logic that seems too much effort at this stage.

bernhard-herzog commented:
I've tweaked the newline handling a bit so that the output should be easier to read. In particular, the "Loading book" lines are now on separate lines

cintek reviewed (approved):
Works
